concepts/objects/about.md

```diff
@@ -0,0 +1,251 @@
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.greeting;
+// => hello world
+
+obj['greeting'];
+// => hello world
+
+// Bracket notation also works with variables.
+const key = 'greeting';
+obj[key];
+// => hello world
+```
+
+Using the dot notation as a short-hand has the same restriction as omitting the quotation marks.
+It only works if the key follows the identifier naming rules.
+
+If you try to retrieve a key that does not exist in the object, JavaScript returns `undefined`.
+See the [Null and Undefined Concept][concept-null-undefined] for more details on this.
+
+You can chain the keys if you want to retrieve a value from a nested object.
+
+```javascript
+const obj = {
+  address: {
+    street: 'Trincomalee Highway',
+    city: 'Batticaloa',
+  },
+};
+
+obj.address.city;
+// => 'Batticaloa'
+
+obj['address']['city'];
+// => 'Batticaloa'
+```
+
+## Adding or Changing a Value
+
+You can add or change a value using the assignment operator `=`.
+Again, there is dot and bracket notation available.
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.greeting = 'Hi there!';
+obj['greeting'] = 'Welcome.';
+
+obj.newKey1 = 'new value 1';
+obj['new key 2'] = 'new value 2';
+
+const key = 'new key 3';
+obj[key] = 'new value 3';
+```
+
+## Deleting an Entry
+
+You can delete a key-value pair from an object using the `delete` keyword.
+
+```javascript
+const obj = {
+  key1: 'value1',
+  key2: 'value2',
+};
+
+delete obj.key1;
+delete obj['key2'];
+```
+
+Note that although `undefined` is returned for non-existing keys, setting a key to `undefined` does not remove the entry.
+Always use `delete` instead.
+
+## Checking Whether a Key Exists
+
+You can check whether a certain key exists in an object with the `hasOwnProperty` method.
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.hasOwnProperty('greeting');
+// => true
+
+obj.hasOwnProperty('age');
+// => false
+```
+
+`hasOwnProperty` returns `false` for [inherited keys][concept-inheritance]. This is usually the desired behavior.
+If you want include inherited keys in the existence check you can use the [`in` operator][mdn-in-operator] instead.
+
+## Looping Through an Object
+
+There is a special `for...in` loop to iterate over all keys of an object.
+
+```javascript
+const obj = {
+  name: 'Ali',
+  age: 65,
+};
+
+for (let key in obj) {
+  console.log(key, obj[key]);
+}
+// name Ali
+// age 65
+```
+
+It might seem like `for...in` always visits the keys in the order in which they appear or were added to the object (insertion order).
+Nevertheless you should not rely on this because there are special cases where the order will differ from the insertion order.
+Always write your code as if the order would be arbitrary.
+If you need a reliable order, use a [`Map`][jsinfo-map] or a nested [array][concept-arrays] instead of an object.
+
+The for-in loop has a pitfall in case the object has [inherited keys][concept-inheritance].
```

Comment:
```suggestion
The `for…in` loop has a pitfall in case the object has [inherited keys][concept-inheritance].
```

Reply:
Not sure which one is the common name, but I think consistency might be nicer. Same applies to the introduction.